src/ast/passes/builtins.cpp

```diff
@@ -106,6 +107,26 @@ std::optional<Expression> Builtins::visit(Call &call)
         return ast_.make_node<Integer>(str->loc, signal_num);
       }
     }
+  } else if (call.func == "__builtin_kfunc_avail") {
+    auto *probe = dynamic_cast<Probe *>(top_level_node_);
+    if (!probe) {
+      return std::nullopt;
+    }
+    if (call.vargs.size() != 1) {
+      call.addError() << "__builtin_kfunc_avail expects 1 argument";
+    } else {
+      String *kfunc = call.vargs.at(0).as<String>();
+      if (!bpftrace_.feature_->has_kfunc_test_insns(kfunc->value)) {
```

Comment:
What is the minimal kernel version here?

Reply:
The `has_kfunc_test_insns` function is determined by us and is independent of the kernel version. For example, if we register a test insns retrieval function for this kfunc in our scenario, `has_kfunc_test_insns` will return `true`.

Reply:
Yes, if some one use `__builtin_kfunc_avail("A")`, there must be a `bpf_insn(s)` get function like `bpf_task_from_pid_insns()`. becuase there are lots of kfuncs, thus, We only need to implement the `bpf_insn` we need, like `bpf_task_from_pid_insns()`.